internal/controller/fluxinstance_controller.go

```diff
@@ -230,12 +247,17 @@ func (r *FluxInstanceReconciler) fetch(ctx context.Context,
 		ctxPull, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 		defer cancel()
 
-		artifactDigest, err := builder.PullArtifact(ctxPull, artifactURL, tmpDir)
-		if err != nil {
+		// Create a keychain for the distribution artifact.
+		if keyChain, err := GetDistributionKeychain(ctxPull, r.Client, obj); err != nil {
 			return "", "", err
+		} else {
```

Comment:
I'd like us to use the more Go idiomatic way and not use an `else` here, just put the line `keyChain, err := GetDistributionKeychain(ctxPull, r.Client, obj)` outside the if condition and the early return for errors after that please.

Reply:
done